pkg/schemadsl/compiler/translator.go

```diff
@@ -434,6 +437,70 @@ func extractTypeAnnotations(typeAnnotationNode *dslNode) ([]string, error) {
 	return annotations, nil
 }
 
+func addTypeAnnotationsToMetadata(relation *core.Relation, typeAnnotations []string) error {
+	if relation.Metadata == nil {
+		relation.Metadata = &core.Metadata{}
+	}
+
+	// Find existing PERMISSION RelationMetadata and update it, or create new one
+	for i, metadataAny := range relation.Metadata.MetadataMessage {
```

Comment:
Move this logic into https://github.com/authzed/spicedb/blob/main/pkg/namespace/metadata.go and follow a similar style to there